Lib/test/test_venv.py

```diff
@@ -360,6 +361,26 @@ def test_multiprocessing(self):
             'pool.terminate()'])
         self.assertEqual(out.strip(), "python".encode())
 
+    def test_deactivate_with_strict_bash_opts(self):
+        bash = shutil.which("bash")
+        if bash is None:
+            self.skipTest("bash required for this test")
+        rmtree(self.env_dir)
+        builder = venv.EnvBuilder(clear=True)
+        builder.create(self.env_dir)
+        activate = os.path.join(self.env_dir, self.bindir, 'activate')
+        test_script = os.path.join(self.env_dir, 'test_strict.sh')
+        with open(test_script, 'w') as f:
+            f.write("""
+set -euo pipefail
+source {}
+deactivate
+            """.format(activate))
+        out, err = check_output([bash, test_script])
+        self.assertEqual(out, "".encode())
```

Comment:
I was going to complain about the inconsistent use of quotes... but it turns out this is actually somewhat consistent with the rest of the file.